test/spec/modules/liveIntentExternalIdSystem_spec.js

```diff
@@ -470,4 +476,100 @@ describe('LiveIntentExternalId', function() {
+    liveIntentExternalIdSubmodule.getId(configWithActivateHoldoutGroup).callback(() => {});
+    expect(window.liModuleEnabled).to.eql(false)
+    expect(window.liTreatmentRate).to.eql(DEFAULT_TREATMENT_RATE)
+  });
+
+  it('should decode values when setting the treatmentRate is undefined', function() {
+    const result = liveIntentExternalIdSubmodule.decode({ nonId: 'foo', vidazoo: 'bar', segments: ['tak'] }, { params: defaultConfigParams });
+    expect(result).to.eql({'lipb': {'lipbid': 'foo', 'nonId': 'foo', 'vidazoo': 'bar', 'segments': ['tak']}, 'vidazoo': {'id': 'bar', 'ext': {'provider': 'liveintent.com'}}});
+    expect(window.liModuleEnabled).to.eql(true)
+    expect(window.liTreatmentRate).to.eql(1)
+  });
+
+  it('should decode values when setting the activateHoldoutGroup parameter', function() {
+    const configWithActivateHoldoutGroup = { ...defaultConfigParams, params: { ...defaultConfigParams.params, activateHoldoutGroup: true } }
+    const result = liveIntentExternalIdSubmodule.decode({ nonId: 'foo', vidazoo: 'bar', segments: ['tak'] }, configWithActivateHoldoutGroup);
+    expect(result).to.eql({'lipb': {'lipbid': 'foo', 'nonId': 'foo', 'vidazoo': 'bar', 'segments': ['tak']}, 'vidazoo': {'id': 'bar', 'ext': {'provider': 'liveintent.com'}}});
+    expect(window.liModuleEnabled).to.eql(true)
+    expect(window.liTreatmentRate).to.eql(DEFAULT_TREATMENT_RATE)
+  });
+
+  it('should decode values with a treatmentRate that is different to the DEFAULT_TREATMENT_RATE', function() {
+    window.liTreatmentRate = 0.7
+    const result = liveIntentExternalIdSubmodule.decode({ nonId: 'foo', vidazoo: 'bar', segments: ['tak'] }, { params: defaultConfigParams });
+    expect(result).to.eql({'lipb': {'lipbid': 'foo', 'nonId': 'foo', 'vidazoo': 'bar', 'segments': ['tak']}, 'vidazoo': {'id': 'bar', 'ext': {'provider': 'liveintent.com'}}});
+    expect(window.liModuleEnabled).to.eql(true)
+    expect(window.liTreatmentRate).to.eql(0.7)
+  });
+
+  it('should NOT decode values when rolling the dice disables the module', function() {
+    randomStub.returns(1)
+    const configWithActivateHoldoutGroup = { ...defaultConfigParams, params: { ...defaultConfigParams.params, activateHoldoutGroup: true } }
+    const result = liveIntentExternalIdSubmodule.decode({ nonId: 'foo', vidazoo: 'bar', segments: ['tak'] }, configWithActivateHoldoutGroup);
+    expect(result).to.eql({});
+    expect(window.liModuleEnabled).to.eql(false)
+    expect(window.liTreatmentRate).to.eql(DEFAULT_TREATMENT_RATE)
+  });
+
+  it('getId and decode should set the global variables correctly when setting the treatmentRate is undefined', function() {
+    const result = liveIntentExternalIdSubmodule.decode({ nonId: 'foo', vidazoo: 'bar', segments: ['tak'] }, { params: defaultConfigParams });
+    liveIntentExternalIdSubmodule.getId(defaultConfigParams).callback(() => {});
```

Comment:
Nice test! You can change it a bit so it reflects the actual flow in Prebid. First `getId` and then `decode`.